docs/reference/modules/asset.md

```diff
@@ -0,0 +1,205 @@
+
+### `extensions`
+The `extensions` option allows the addition of custom methods for processing the Javascript and SCSS in the `ui/src` folder. The schema used within the option conforms to that used for configuring webpack, which you can learn more about [here](https://webpack.js.org/configuration/). This property can take either a simple object with a name property or a function that returns an object. 
+
+#### `extensions` simple object
+
+#### Example
+
+Modification of the `resolve` property of the webpack configuration using a simple object. New values will be merged with existing values:
+
+<AposCodeBlock>
+
+```javascript
+const path = require("path");
+
+module.exports = {
+ // ...
+ webpack: {
+   extensions: {
+     modifyResolve: {
+       resolve: {
+         extensions: [ '.jsx' ],
+         alias: {
+           'Utilities': path.join(process.cwd(), 'lib/utils')
+         }
+       }
+     }
+   }
+ },
+ // ...
+};
+
+```
+<template v-slot:caption>
+modules/my-module/index.js
+</template>
+</AposCodeBlock>
+
+::: note
+Make sure the new values are of the same type as the existing values, or they will replace the existing values, e.g., the webpack configuration `extensions` property takes an array, whereas `alias` takes an object.
+:::
+
+This will add both processing for files with an extension of `.jsx` and expose a shortcut to the `lib/utils` folder that we can use in our frontend files. For example, to import a custom method from our `methods.js` library in this folder.
+
+<AposCodeBlock>
+
+```javascript
+import  { customMethod } from ('Utilities/methods.js');
+
+export default () => {
+// Code utilizing customMethod
+};
+```
+<template v-slot:caption>
+modules/my-module/ui/src/index.js
+</template>
+</AposCodeBlock>
+
+Adding a unique name for the object within the apostrophe `extensions` property is important. Passing an object with the same name to an `extensions` property of another module will cause a conflict. The configuration file will be set to the `extensions` object in the last loaded module.
+
+#### `extensions` function
+
+Functions in the `extensions` property take an `option` parameter to allow multiple modules to modify the same extension through the [`extensionOptions`](#extensionoptions) property. Each property to be altered across multiple modules must be present in the option that was originally returned.
+
+<AposCodeBlock>
+
+```javascript
+const path = require("path");
+
+module.exports = {
+ // ...
+  webpack: {
+   extensions: {
+     addAlias(options) {
+       return {
+         resolve: {
+           extensions: [ ...(options.extensions || []), '.jsx' ],
+           alias: {
+             ...(options.alias || {}),
+             Utilities: path.resolve(process.cwd(), './lib/utils')
+           }
+         }
+       };
+     }
+   }
+ },
+ // ...
+};
+
+```
+<template v-slot:caption>
+modules/my-module/index.js
+</template>
+</AposCodeBlock>
+
+### `extensionOptions`
```

Comment:
Any module can pass options to an existing webpack `extension` as long as it was defined as a function accepting an `options` object.

The `extensionOptions` property is an object, with a sub-property matching each extension it wishes to pass options to.

The values for each sub-property can be simple objects, in which case they merge into the options object to be passed to that extension, or functions, in which case they take an `options` object containing the options so far and must return a new object with any desired changes made.

Reply:
CHanged.